docs/deprecation-guide.md

```diff
@@ -0,0 +1,139 @@
+# When and how to deprecate hooks and non-private functions in subscriptions-core
+
+When breaking changes are made to a filter, action or non-private function in subscriptions-core, we need to continue to support the current behaviour of it while encouraging downstream extensions or themes to use the updated version.
+
+When we mark a function or hook as deprecated, a deprecation notice warning will display when the function or hook is used.
+
+-   [Filters](#filters)
+-   [Actions](#actions)
+-   [Non-private functions](#non-private-functions)
+
+## Identifying functions needing deprecation
+
+-   The type of one or more function parameters is changing.
+-   A function parameter has been added or removed.
+-   The method or function is non-private (e.g. `public function ...`).
+
+## Filters
+
+The deprecation of filters is handled where the filter is triggered, aka `apply_filter`.
+
+In the example below, the `$post` parameter is being replaced with `$order`.
+
+**Before**
+
+```php
+$orders_to_display = apply_filters( 'woocommerce_subscriptions_admin_related_orders_to_display', $orders_to_display, $subscriptions, $post );
+```
+
+**After**
+
+```php
+// Wrap the legacy call to apply_filter in a has_filter statement
```

Comment:
What's the rationale for `has_filter` wrap? I understand there are good reasons, let's document the why then the how (in case the how changes; the why will likely stay constant).